### PR TITLE
sentry: add PKCS8 support

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

# Description

Add PKCS8 support.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #3240

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by daixiang0** [comment]
@artursouza please help review

File: pkg/sentry/csr/csr.go [hunk 1]
```diff
@@ -212,3 +213,81 @@ func newSerialNumber() (*big.Int, error) {
 	}
 	return serialNum, nil
 }
+
+type CredentialRequests struct {
+	PrivateKey  *certs.PrivateKey
+	Certificate *x509.CertificateRequest
+}
+
+func PEMCredentialRequestsFromFiles(certPem, keyPem []byte) (*CredentialRequests, error) {
```
Comment (artursouza) [inline pkg/sentry/csr/csr.go:222]: Where is this being used apart from test?

Comment (daixiang0) [reply pkg/sentry/csr/csr.go]: Remove it.

File: pkg/sentry/csr/csr.go [hunk 2]
```diff
@@ -212,3 +212,38 @@ func newSerialNumber() (*big.Int, error) {
 	}
 	return serialNum, nil
 }
+
+type CredentialRequests struct {
```
Comment (artursouza) [inline pkg/sentry/csr/csr.go:216]: It seems to be unused.

File: pkg/sentry/csr/csr.go [hunk 3]
```diff
@@ -212,3 +212,38 @@ func newSerialNumber() (*big.Int, error) {
...
+	PrivateKey  *certs.PrivateKey
+	Certificate *x509.CertificateRequest
+}
+
+func DecodePEMCertificateRequests(rest []byte) ([]*x509.CertificateRequest, error) {
```
Comment (artursouza) [inline pkg/sentry/csr/csr.go:221]: Unused too.

File: pkg/sentry/csr/csr.go [hunk 4]
```diff
@@ -212,3 +212,38 @@ func newSerialNumber() (*big.Int, error) {
...
+		}
+		if cert != nil {
+			// it's a cert, add to pool
+			certs = append(certs, cert)
+		}
+	}
+	return certs, nil
+}
+
+func decodeCertificateRequestPEM(rest []byte) (*x509.CertificateRequest, []byte, error) {
```
Comment (artursouza) [inline pkg/sentry/csr/csr.go:239]: This will also be unused.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: pkg/sentry/csr/csr.go (hunk 4 above)
Comment (daixiang0) [reply pkg/sentry/csr/csr.go]: Seems should better enable deadcode linter.

**Review by artursouza — APPROVED** [review]

**Comment by daixiang0** [comment]
@artursouza all tests pass.

**Review by artursouza — APPROVED** [review]

**Pull request merged**